cmake/protobuf_generate_cpp.cmake

```diff
@@ -0,0 +1,164 @@
+# This file declares functions adding custom commands for generating C++ files from *.proto files:
+# function (protobuf_generate_cpp SRCS HDRS)
+# function (protobuf_generate_grpc_cpp SRCS HDRS)
```

Comment:
I moved all protobuf-related generating code into this file.

`protobuf_generate_grpc_cpp()` works slightly different now: it generates all four files `*.pb.h`, `*.pb.cc`, `*.grpc.pb.h`, `*.grpc.pb.h`.